src/main/src/main/java/org/geoserver/catalog/Predicates.java

```diff
@@ -198,6 +198,17 @@ public static Filter fullTextSearch(final String subsequence) {
         return contains(ANY_TEXT, subsequence);
     }
 
+    /**
+     * Returns a predicate for exact match search (case-insensitive) against the <code>AnyText
+     * </code> property with the provided term.
+     *
+     * @param term the exact term to search for, ignoring case
+     * @return the filter for exact match search
+     */
+    public static Filter exactTermSearch(final String term) {
+        return factory.like(ANY_TEXT, fixSpecials(term));
```

Comment:
Why use "like", it seems you're looking for simple equality... if so, ``Predicates.equal`` should also work.

Reply:
Equal can also run in case-insensitive mode, but Predicates.equal is forcing case sensitiveness inside. Simple equality is faster, like is implemented as regular expressions, but it should not be a big deal. You can either expose a Predicates.equal variant that allows to tun the matchCase attribute, or keep it this as it is.